### ISSUE TITLE
Remove gcp-pilot dependency and widen python support

### ISSUE THREAD

**Issue by thclark** [octue/django-gcp#52]
### Problem

t present, poetry.lock is out of sync with pyproject.toml. 

Doing `poetry lock --no-update` to bring it back in sync reveals that there's an incompatibilty between the allowed version range of `gcp-pilot` and our allowed python range.

We need to go either to v1 or way back to a much earlier version (0.18) than we're currently using (0.40.3). In v1 as it stands, they only support python 3.11.

Basically, even though it's maintained, gcp-pilot is a clusterfuck of dependency management, it's impossible to treat it as a stable library. We should harvest the chunks of code we need from 0.40.3 and reimplement the bits we need ourselves (mostly only used for tasks).

In #44 we wish to update dependencies to python 3.12 which won't be possible until this is resolved.
